webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/SetupRedirect.scala

```diff
@@ -38,36 +38,45 @@
 package com.normation.rudder.web.snippet
 
 import bootstrap.liftweb.RudderConfig
-import com.normation.box.*
 import com.normation.rudder.AuthorizationType.Administration
 import com.normation.rudder.users.CurrentUser
+import com.normation.zio.UnsafeRun
 import net.liftweb.common.*
 import net.liftweb.http.CurrentReq
 import net.liftweb.http.DispatchSnippet
 import net.liftweb.http.js.JsCmd
 import net.liftweb.http.js.JsCmds.*
 import scala.xml.NodeSeq
+import zio.syntax.*
 
 class SetupRedirect extends DispatchSnippet with Loggable {
 
-  private val configService = RudderConfig.configService
+  private val configService         = RudderConfig.configService
+  private val pluginSettingsService = RudderConfig.pluginSettingsService
 
   def dispatch: PartialFunction[String, NodeSeq => NodeSeq] = {
     case "display" => _ => WithNonce.scriptWithNonce(Script(display()))
   }
 
   def display(): JsCmd = {
-
-    configService.rudder_setup_done().toBox match {
-      case Full(false)
-          if CurrentUser.checkRights(Administration.Write) && !CurrentReq.value.request.url.contains("administration/setup") =>
+    (for {
+      isSetupDone     <- configService.rudder_setup_done()
+      setupFileExists <- pluginSettingsService.settingsFileExists()
```

Comment:
this check should not be done each time. It's costlier than a setting check, and should only be done in the case the setting service answer "not done".

Reply:
yes ! I used a flatMap instead, and in fact I could remove the `settingsFileExists` check, it can be superseded with the more powerful check on the settings being defined